compiler-core/src/error.rs

```diff
@@ -1843,12 +1843,22 @@ constructing a new record with its values."
                     variables,
                     name,
                     type_with_name_in_scope,
+                    situation
                 } => {
                     let text = if *type_with_name_in_scope {
                         wrap_format!("`{name}` is a type, it cannot be used as a value.")
                     } else {
-                        wrap_format!("The name `{name}` is not in scope here.")
+                        let mut situation_text = wrap_format!("The name `{name}` is not in scope here.");
```

Comment:
We now have 2 fields that describe the situation, that being `type_with_name_in_scope` and `situation`. Let's merge them together by making `type_with_name_in_scope` a variant of the `situation` enum.